src/util/use-config.tsx

```diff
@@ -0,0 +1,65 @@
+import Store from "electron-store"
+import React, {
+  useState,
+  useEffect,
+  createContext,
+  ReactNode,
+  useContext,
+} from "react"
+
+/**
+ * This module uses shared context to wrap and subscribe to electron-config
+ */
+
+interface IConfigType {
+  siteTabs: Array<string>
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const ConfigContext = createContext<Store<IConfigType> | undefined>(undefined)
+/**
+ * Wraps the site root element in gatsby-browser
+ */
+export function ConfigProvider({
+  children,
+}: {
+  children: ReactNode
+}): JSX.Element {
+  return (
+    <ConfigContext.Provider
+      value={typeof window !== `undefined` ? new Store() : undefined}
```

Comment:
do we ssr this at all? Cause this will be majorly problematic if we do.

Reply:
Yes, we do! The whole app is served from a built app.

Reply:
To expand, this is ok because we gracefully handle missing config stores in the hook below. All the config values can be undefined, so we explicitly handle that case everywhere we use them.

Reply:
Fascinating. Would love to see some code comments to explain because this just screams SSR nightmare to me